database.cc

```diff
@@ -2321,6 +2321,71 @@ future<> database::drain() {
     co_await _commitlog->shutdown();
 }
 
+future<std::unordered_map<sstring, std::unordered_map<std::pair<sstring, sstring>, db::snapshot_ctl::snapshot_details, utils::tuple_hash>>> database::get_snapshot_details() {
+    // We require a seastar thread to run `lister::scan_dir`s
+    co_return co_await seastar::async([this] () {
+        std::vector<sstring> data_dirs = _cfg.data_file_directories();
+        auto dirs_only_entries_ptr =
```

Comment:
Why does it have to be allocated on a heap? Can't it be stack allocated?

Reply:
Like details

Reply:
Hm, I based this part on other places in our codebase where we search through the directories and they all use a pointer. But tbh now that you mention it, I have no idea why. Changed it to automatic lifetime and it seems to be working fine